### ISSUE TITLE
Syncing Stops and/or Messes Up

### ISSUE THREAD

**Issue by ReturnMe** [Natizyskunk/vscode-sftp#67]
**Do you read the FAQ?**
- Yes.

**Describe the bug**
In total, 167 MB.
There are 2390 files and 250 folders.
Is it a problem because it's too much?
The problem is; I choose local from remote, but after a while, a note gets stuck.
It then says that it is from local to remote, in the lower left.

The same thing happens even if I choose Sync Local -> Remote.
Exactly the same happens even if I choose Sync Both Directions.

**To Reproduce**
1. Files panel, right click
2. Sync Remote -> Local
3. Not see error. So it waits for minutes. However, the file; 300kb. And my connection is very good.

**Expected behavior**
Download, jqvmap.html file **(Remote -> Local)**

**Screenshots**
![image](https://user-images.githubusercontent.com/17087093/136477100-1f843ee6-c653-4dea-8fbe-4ebe241a0175.png)

**Desktop (please complete the following information):**
 - OS: Win x64
 - VSCode Version 1.61.0
 - Extension Version 1.15.4

**Extension Logs from Startup** - *required*
[SFTP Debug.txt](https://github.com/Natizyskunk/vscode-sftp/files/7307517/SFTP.Debug.txt)

**Comment by ReturnMe** [comment]
There are new findings on this.
I noticed it myself when I was making plugins for myself.
I think;
When the file is downloaded, the time of the downloaded file is new.
When it is uploaded, the time of the uploaded file becomes new.
When this is the case, it is constantly trying to transfer a new one. And I guess it goes into an endless loop.
However, the change times of the files should have been the same.
I guess copying the file instead of recreating it might be the solution. Or I don't know :)

By the way, how soon can you complete this? I still tremendously need SFTP.

**Comment by Natizyskunk** [comment]
@ReturnMe,

I've actually take a look at it but i can't reproduce your error...
For me everything work fine and all files/folders are correctly uploaded.

Maybe someone can help on this.

**Comment by ReturnMe** [comment]
> @ReturnMe,
> 
> I've actually take a look at it but i can't reproduce your error... For me everything work fine and all files/folders are correctly uploaded.
> 
> Maybe someone can help on this.

I was getting this error frequently.
I'll try again and post my results on that here.

**Comment by ReturnMe** [comment]
Yep I had to use it again and a project with 8400 files and 900 directories hangs when selecting "both".

And, It corrupts filenames containing Turkish characters.

In summary, the problem remains the same. Same system, same everything.

**Comment by Natizyskunk** [comment]
@ReturnMe, Thanks for your updates.
As for now, i sadly don't have any option to fix your problem.
The best i can do for now is to keep this issue open and hope that someone come with the same problem or a solution (even better).

I'm sorry that i can't help you more.

**Comment by Natizyskunk** [comment]
Related issues : #8, #67, #67, #150, #196

**Comment by chobao** [comment]
> Yep I had to use it again and a project with 8400 files and 900 directories hangs when selecting "both".
> 
> And, It corrupts filenames containing Turkish characters.
> 
> In summary, the problem remains the same. Same system, same everything.

I fix the infinite waiting of sftp by finding a broken link file in my directory and adding it to the `ignore` in `sftp.json`.
